### ISSUE TITLE
[8.5.5] gradient exploding이 RNN의 어떤 과정에서 발생할 수 있나요?

### ISSUE THREAD

**Issue by 1gyeol-KIM** [woowacampers/d2l-pytorch#4]
RNN에서 gradient exploding이 발생할 수 있기 때문에, gradient clipping(어떤 임계값을 넘지 않도록 기울기 값을 자름)을 수행함으로써 문제를 fix할 수 있다고 이해했는데요.

어떤 과정에서 기울기가 폭주? 할 가능성이 생기는지 궁금합니다. 제가 놓친 부분이 있는 것 같아요...

**Comment by 1gyeol-KIM** [comment]
8.7. Backpropagation Through Time
에서 설명이 되어 있는 것 같아요. 
<img width="746" alt="스크린샷 2021-03-03 오후 3 07 20" src="https://user-images.githubusercontent.com/42639690/109760524-28398780-7c32-11eb-9aeb-2d3d7e858ad6.png">
식 (8.7.15)의 model parameter matrix 𝐖⊤ℎℎ 가 매우 클 때? (power를 크기로 해석하면 될까요) eigenvalue 값에 따라 gradient vanishing / exploding problem이 발생할 수 있을 것 같습니다.
gradient clipping이 exploding을 빠르게 fix할 수 있는 테크닉들 중에 하나라고 생각합니다.

**Comment by staycozyboy** [comment]
![image](https://user-images.githubusercontent.com/37537248/109766129-1b6c6200-7c39-11eb-937f-88224589f8fb.png)

[norm의 성질에 따라 성립하는 부등호 증명](https://imgur.com/zcuKkv2)

**Comment by staycozyboy** [comment]
![image](https://user-images.githubusercontent.com/37537248/109766736-ead8f800-7c39-11eb-9af2-5e8df6d9f7a1.png)

[eigen value에 대한 증명](https://imgur.com/zcuKkv2)

**Comment by staycozyboy** [comment]
### 추가적인 질문
1. eigen value 증명에 대한 이해
2. 중간에 주어진 부등호만 고려했을 때, W_hh의 norm이 1보다 클 때, gradient exploding이 일어나는 것을 보장할 수 있는 것인지 잘 모르겠습니다.

https://ratsgo.github.io/deep%20learning/2017/10/10/RNNsty/
